src/Compilers/CSharp/Portable/Symbols/TypeSymbolExtensions.cs

```diff
@@ -1361,6 +1362,64 @@ public static bool IsFileTypeOrUsesFileTypes(this TypeSymbol type)
             return foundType is not null;
         }
 
+        internal static string? AssociatedFileIdentifier(this NamedTypeSymbol type)
+        {
+            if (type is not SourceMemberContainerTypeSymbol { AssociatedSyntaxTree: SyntaxTree tree })
```

Comment:
>not SourceMemberContainerTypeSymbol [](http://example.com/codeflow?start=24&length=35)

Do we need to handle substituted types? If not, consider asserting `type.IsDefinition`.

Reply:
I found that we access this member to build metadata names whether the type is a definition or not. One place where we end up doing this is in `TypeSymbolExtensions.IsSpanOrReadOnlySpanChar`. I think it makes sense to still append the prefix in the same way even if a generic file type is constructed with some arguments, for example, so I will refrain from asserting `IsDefinition` here. `INamedTypeSymbol.MetadataName` is also a public API.